### DISCUSSION TITLE
gRPC Error when running locally

### DISCUSSION THREAD

**Discussion by stephan-au** [fzyzcjy/flutter_convenient_test#360] [Q&A]
When I run the GUI on Windows and run the integration tests I get the exception:
`E/flutter ( 2515): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: gRPC Error (code: 14, codeName: UNAVAILABLE, message: Error connecting: SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 127.0.0.1, port = 37430, details: null, rawResponse: null, trailers: {})`

It doesn't run the tests and I don't see the tests inside the GUI.

I can get it to work when I have flutter_convenient_test package locally on my pc and put it in dependency_overrides in my pubspec.yaml.

But I want it to work with the dependency the normal way, otherwise I need to push the whole flutter_convenient_test package to the CICD to build my project.

Things I tried:

- Set environment var - Convenient_test_manager_host = 10.0.2.2
- Set default value of kConvenientTestManagerHost in const.dart to 10.0.2.2
- Disable firewall
- Disable anti-virus
- Downloaded the latest GUI version from GIT
- Running the GUI in administrator mode
- Flutter clean + flutter pub get

Any idea what I am missing?

**Comment by fzyzcjy** [comment]
> I can get it to work when I have flutter_convenient_test package locally on my pc and put it in dependency_overrides in my pubspec.yaml.

Hmm, that's weird. Are you using the latest remote dependencies? From the history https://github.com/fzyzcjy/flutter_convenient_test/commits/master/, it seems that 1.3.2 is same as master (except for some minor changes unrelated to this)

**Comment by ghost** [comment]
@fzyzcjy If it helps any, I get this error when I don't have the manager running and then attempt `flutter run myapp/integration_test/main_test.dart`

If I open the manager first and then run the tests this error goes away 🤔 

Not sure if this is a bug that should be reported or if tests should be able to run without the manager open?

**Reply by fzyzcjy** [reply]
> I get this error when I don't have the manager running

IMHO you need the manager (either the GUI, or the executable binary), since there are some features that may not exist without manager (e.g. isolation mode).

**Reply by ghost** [reply]
Ok great, will put that bit of info in the documentation when I make changes so people are aware :)
